portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -133,14 +166,23 @@ private String getParameter(HttpServletRequest request, String name, String defa
    */
   class Comment {
     // The fields that hold the relevant comment data.
+    private long id;
     private String text;
     private Date date;
 
+    // No ID is required when creating Comment Entity.
     public Comment(String text, Date date) {
       this.text = text;
       this.date = date;
     }
 
+    // ID is required when creating Comment object for frontend.
+    public Comment(long id, String text, Date date) {
```

Comment:
These two constructors look pretty similar but iiuc they're used in pretty different situations, ie: either to construct a new entity or to express a pre-existing entity. The comments help clarify this, but you could also consider using different arguments/method signatures to help emphasize the difference. Eg: this constructor could accept an Entity object.

Definitely not a required change, and feel free to address this in a separate CL if at all.

Reply:
To tack onto Tristan's suggestion. In situations where you have the same method signatures, but with just a few omitted, a very common practice (and one I'd recommend) is to do this:

public Comment(String text, Date date) {
   this(0, text, date);
}

public Comment(long id, String text, Date date) {
      this.id = id;
      this.text = text;
      this.date = date;
    }

Reply:
That makes sense, I hadn't heard of that technique before @bheff-work -- thanks!